src/storage.rs

```diff
@@ -254,6 +263,21 @@ impl<T> UnprotectedStorage<T> for VecStorage<T> {
     }
 }
 
+/// A dummy storage type, used for cases where the component
+/// doesn't contain any data and instead works as a simple flag.
+pub struct DummyStorage<T>(T);
+
+impl<T: Clone + Default> UnprotectedStorage<T> for DummyStorage<T> {
```

Comment:
Why? Clone is less restrictive and enough for my needs.

> On Apr 28, 2016, at 0:33, CSherratt <notifications@github.com> wrote:
> 
> In src/storage.rs:
> 
> > @@ -254,6 +263,21 @@ impl<T> UnprotectedStorage<T> for VecStorage<T> {
> >      }
> >  }
> >  
> > +/// A dummy storage type, used for cases where the component
> > +/// doesn't contain any data and instead works as a simple flag.
> > +pub struct DummyStorage<T>(T);
> > +
> > +impl<T: Clone + Default> UnprotectedStorage<T> for DummyStorage<T> {
> T should probably be a Copy type over a clone.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>